File: sleet-api-service/src/main/java/com/sleet/api/domain/OptionChain.java
package com.sleet.api.domain;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.List;
import java.util.Map;

/**
 * Represents an option chain for a symbol consumed from the TD API
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class OptionChain {

    private String symbol;
    private double interestRate;
    private double underlyingPrice;
    private double volatility;
    private Map<String, Map<String, List<Option>>> callExpDateMap;
    private Map<String, Map<String, List<Option>>> putExpDateMap;

    public String getSymbol() {
        return symbol;
    }

    public double getInterestRate() {
        return interestRate;
    }

    public double getUnderlyingPrice() {
        return underlyingPrice;
    }

    public double getVolatility() {
        return volatility;
    }

    public Map<String, Map<String, List<Option>>> getCallExpDateMap() {
        return callExpDateMap;
    }

    public Map<String, Map<String, List<Option>>> getPutExpDateMap() {
        return putExpDateMap;
    }

    @Override
    public String toString() {
        return getSymbol() + " : " + getUnderlyingPrice();
    }
}

File: src/main/java/com/sleet/api/model/OptionChain.kt
package com.sleet.api.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

/**
 * Represents an option chain for a symbol consumed from the Schwab API
 *
 * @author mautomic
 */
@JsonIgnoreProperties(ignoreUnknown = true)
open class OptionChain {
    val symbol: String? = null
    val interestRate = 0.0
    val underlyingPrice = 0.0
    val volatility = 0.0
    var callExpDateMap: Map<String, Map<String, List<Asset>>>? = null
    var putExpDateMap: Map<String, Map<String, List<Asset>>>? = null
    override fun toString(): String {
        return "$symbol : $underlyingPrice"
    }
}